### PR TITLE
Update external toolchains to support MSVC and Apple targets.

### PR DESCRIPTION
**Pull request by Alexhuszagh**
Adds support for building custom MSVC and Apple Darwin targets, and references these images in the README.

### CODE REVIEW

**Comment by Alexhuszagh** [comment]
Since this doesn't change anything in the repository, I didn't update the changelog. I've also confirmed all the images compile, that the x86 targets have valid runners, that the ARM64 image compiles correctly, and that the ARM image (which doesn't support `std` currently) compiles a `no_std` library.

**Review by Emilgardis — APPROVED** [review]

**Comment by Alexhuszagh** [comment]
I'm going to hold off on this for a short period until I get macOS working, improve lowercase handling slightly, and add to the documentation that by default they should update the `cross-toolchains` repository to the latest version by default, so we only need to increment the submodule when very important targets are added.

File: README.md [hunk 1]
```diff
@@ -393,6 +393,7 @@ terminate.
 due to a [compiler bug](https://github.com/rust-lang/rust/issues/85821), you will have to build them yourself for now.-->
 
 Additional Dockerfiles for other targets can be found in [cross-toolchains](https://github.com/cross-rs/cross-toolchains).
+These include MSVC and Apple Darwin targets, which for legal reasons we cannot ship pre-built images of.
```
Comment (Emilgardis) [inline README.md:396]:
```suggestion
These include MSVC and Apple Darwin targets, which we cannot ship pre-built images of.
```

**Comment by Alexhuszagh** [comment]
bors r=Emilgardis

**Pull request merged**